include/aws/common/atomics.inl

```diff
@@ -0,0 +1,56 @@
+#ifndef AWS_COMMON_ATOMICS_INL
+#define AWS_COMMON_ATOMICS_INL
+
+/*
+ * Copyright 2010-2018 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#define AWS_ATOMICS_INLINE
+
+#include <aws/common/atomics.h>
+
+/* Helpers for converting aws_atomic_var into its parts. */
+#define AWS_ATOMIC_VAR_PTRVAL(var) (var->value)
+#define AWS_ATOMIC_VAR_INTVAL(var) (*(aws_atomic_impl_int_t *)var)
+
```

Comment:
I'm not sure this is better than the union for readability or maintenance, what was the impetus for the change?

Reply:
We don't know how big the int type is here.